tests/test_isrFunctions.py

```diff
@@ -0,0 +1,402 @@
+
+        for d, t in zip(defectList, transposed):
+            assert d.getBBox().getDimensions().getX() == t.getBBox().getDimensions().getY()
+            assert d.getBBox().getDimensions().getY() == t.getBBox().getDimensions().getX()
+
+    def test_makeThresholdMask(self):
+        r"""Thresholding.
+
+        Expect list of defects to have elements.
+        """
+        defectList = ipIsr.makeThresholdMask(self.mi, 200, growFootprints=2, maskName='SAT')
+
+        self.assertGreater(len(defectList), 0)
+
+    def test_interpolateFromMask(self):
+        r"""Interpolate mask.
+
+        Expect number of interpolated pixels to be non-zero.
+        """
+        ipIsr.makeThresholdMask(self.mi, 200, growFootprints=2, maskName='SAT')
+        for growFootprints in (0, 2):
+            with self.subTest(growFootprints=growFootprints):
+                interpMaskedImage = ipIsr.interpolateFromMask(self.mi, 2.0,
+                                                              growFootprints=growFootprints, maskName='SAT')
+                numBit = maskCountPixels(interpMaskedImage, "INTRP")
+                self.assertGreaterEqual(numBit, 0)
+
+    def test_saturationCorrection(self):
+        r"""Saturation correction.
+
+        Expect number of mask pixels with SAT marked to be non-zero.
+        """
+        for interpolate in (True, False):
+            with self.subTest(interpolate=interpolate):
+                corrMaskedImage = ipIsr.saturationCorrection(self.mi, 200, 2.0,
+                                                             growFootprints=2, interpolate=interpolate,
+                                                             maskName='SAT')
+                numBit = maskCountPixels(corrMaskedImage, "SAT")
+                self.assertGreaterEqual(numBit, 0)
+
+    def test_trimToMatchCalibBBox(self):
+        r"""Trim to match raw data.
+
+        Expect bounding boxes to match.
+        """
+        darkExp = isrMock.DarkMock().run()
+        darkMi = darkExp.getMaskedImage()
+
+        nEdge = 2
+        newDark = darkMi[nEdge:-nEdge, nEdge:-nEdge, afwImage.LOCAL]
+        darkMi = newDark
+        newInput = ipIsr.trimToMatchCalibBBox(self.mi, darkMi)
+
+        self.assertEqual(newInput.getImage().getBBox(), newDark.getImage().getBBox())
+
+    def test_darkCorrection(self):
+        r"""Dark correction.
+
+        Expect round-trip application to be equal.
+        Expect RuntimeError if sizes are different.
+        """
+        darkExp = isrMock.DarkMock().run()
+        darkMi = darkExp.getMaskedImage()
+
+        image = self.mi.getImage().getArray()
+        ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=True)
+        ipIsr.darkCorrection(self.mi, darkMi, 1.0, 1.0, trimToFit=True, invert=True)
+
+        diff = image - self.mi.getImage().getArray()
+        assert (diff.sum() == 0)
+
+        newDark = darkMi[1:-1, 1:-1, afwImage.LOCAL]
+        darkMi = newDark
```

Comment:
Why not `darkMi = darkMi[1:-1, 1:-1, afwImage.LOCAL]`? Similarly in various places below.